File: app/src/main/java/com/example/newspal/model/ArticleDao.java
package com.example.newspal.model;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ArticleDao {
    @Query("SELECT * FROM Article")
    LiveData<List<Article>> getSavedArticles();

    @Insert
    void saveArticle(Article article);

    @Delete
    void deleteSavedArticle(Article article);
}

File: app/src/main/java/com/example/newspal/data/ArticleDao.kt
package com.example.newspal.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface ArticleDao {

    @Query("SELECT * FROM Article")
    fun getSavedArticles(): LiveData<List<Article>>

    @Insert
    suspend fun saveArticle(article: Article)

    @Delete
    suspend fun deleteArticle(article: Article)

}